### DISCUSSION TITLE
how to define a non case sensitive Data Type Rules

### DISCUSSION THREAD

**Discussion by zrwsmd** [eclipse-langium/langium#1464] [Q&A] [answered]
Real_type_name returns string:
'LREAL' | 'REAL';

how to change this Data Type Rules can match any 'lreal'  or 'real' without validate case sensitive,for example ,'LreAl' also match this rule

**Comment by msujew** [comment]
Hey @zrwsmd,

you can define terminals explicitly for those keywords (note that they should be listed before other, more general terminals):

```
terminal LREAL: /LREAL/i;
terminal REAL: /REAL/i;
```

The `i` makes the corresponding regex case insensitive. In case your whole grammar is case insensitive, you can use the [grammar option](https://github.com/eclipse-langium/langium/blob/26cca7ae2448e5b579244e76e2aeb11bd8550927/examples/arithmetics/langium-config.json#L7) to disable case sensitivitive for all keywords.

**Reply by zrwsmd** [reply]
i can not directly add 'i' at end of rule terminal,
![image](https://github.com/eclipse-langium/langium/assets/26995040/90aa744f-ae7f-4742-b6cf-94fdc29511cc)

**Reply by zrwsmd** [reply]
grammar St
entry St:
 bb=REAL;


terminal REAL: /REAL/;

**Reply by msujew** [reply, marked as answer]
The playground is still running [Langium version `1.2.0`](https://github.com/eclipse-langium/langium-website/blob/823357a00f93267ae139670f77f4e2ba753dfd33/hugo/package.json#L41), while [the feature](https://github.com/eclipse-langium/langium/pull/1070) has been added in 1.3.0 of Langium.
